server/jobs/storage_backend.go

```diff
@@ -0,0 +1,40 @@
+package jobs
+
+import (
+	"io"
+
+	"github.com/runatlantis/atlantis/server/core/config/valid"
+)
+
+//go:generate pegomock generate -m --use-experimental-model-gen --package mocks -o mocks/mock_storage_backend.go StorageBackend
+
+type StorageBackend interface {
+	// Checks the backend storage for the specified key
+	IsKeyExists(key string) bool
+
+	// Read logs from the storage backend. Must close the reader
+	Read(key string) io.ReadCloser
+
+	// Write logs to the storage backend
+	Write(key string, logs []string) (success bool, err error)
```

Comment:
This returns success bool which indicates if the logs were successfully persisted. This would make the code path cleaner when log persistence is not configured. The `NoopStorageBackend` returns `false, nil ` which means the logs were not persisted without any errors. So, we leave the output buffer as it is instead of clearing it up.